xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/DBListClass.java

```diff
@@ -80,7 +81,7 @@ public DBListClass()
             // were empty strings). This means we need to check for NULL and ignore NULL entries
             // from the list.
             if (item != null) {
-                if (item instanceof String) {
+                if (item instanceof String || item instanceof Number) {
                     result.add(new ListItem((String) item));
```

Comment:
This will fail. Marius was suggesting to use this instead:

    result.add(new ListItem(item.toString()));

or:

    result.add(new ListItem(String.valueOf(item)));